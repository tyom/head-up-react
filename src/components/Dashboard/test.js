import React from 'react';
import Dashboard from './';

const scrollIntoViewSpy = jest.fn();

function mountComponent(props) {
  const component = mount(<Dashboard {...props}>content</Dashboard>);
  const componentInstance = component.instance();

  componentInstance._dashboard = {
    scrollIntoView: scrollIntoViewSpy,
  };

  return componentInstance;
}

describe('Dashboard: structure', () => {
  let rendered;

  afterEach(() => {
    expect(rendered).toMatchSnapshot();
  });

  test('default component', () => {
    rendered = render(<Dashboard name="my dashboard">content</Dashboard>);
  });

  test('active state', () => {
    rendered = render(
      <Dashboard name="my dashboard" isActive={true}>
        Active dashboard
      </Dashboard>
    );
  });
});

describe('Dashboard: behaviour', () => {
  let componentInstance;

  describe('scrolling', () => {
    beforeEach(() => {
      componentInstance = mountComponent({
        name: 'my dashboard',
        isActive: true,
      });
    });

    afterEach(() => {
      jest.resetAllMocks();
    });

    test('scroll into view on mount', () => {
      componentInstance.componentDidMount();
      expect(scrollIntoViewSpy).toBeCalledWith(undefined);
    });

    test('scroll into view on update', () => {
      componentInstance.componentDidUpdate();
      expect(scrollIntoViewSpy).toBeCalledWith({ behavior: 'smooth' });
    });
  });

  test('scrolling only for active', () => {
    componentInstance = mountComponent({
      name: 'my dashboard',
      isActive: false,
    });

    componentInstance.componentDidUpdate();
    expect(scrollIntoViewSpy).not.toBeCalled();
  });
});
