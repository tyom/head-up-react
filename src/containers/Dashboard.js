import {connect} from 'react-redux';
import Dashboard from '../components/Dashboard';

const mapStateToProps = (state, ownProps) => ({
  isActive: ownProps.name === state.headup.activeDashboard
});

export default connect(mapStateToProps)(Dashboard);
