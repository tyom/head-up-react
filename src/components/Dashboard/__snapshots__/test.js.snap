// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard: structure active state 1`] = `
<div
  class="style_dashboard__3um69 style_is-active__3ctG-"
  id="my-dashboard"
>
  Active dashboard
</div>
`;

exports[`Dashboard: structure default component 1`] = `
<div
  class="style_dashboard__3um69"
  id="my-dashboard"
>
  content
</div>
`;
